src/shims/unix/fs.rs

```diff
@@ -559,8 +559,14 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 );
             };
 
-            if mode != 0o666 {
-                throw_unsup_format!("non-default mode 0o{:o} is not supported", mode);
+            match this.tcx.sess.target.os.as_ref() {
+                // Support all modes on UNIX host
+                "linux" | "freebsd" | "android" | "macos" => {}
+                // Non-UNIX (aka Windows) only support default mode
+                _ =>
+                    if mode != 0o666 {
+                        throw_unsup_format!("non-default mode 0o{:o} is not supported", mode);
+                    },
```

Comment:
Now you are just completely ignoring the mode. This is *not* what was suggested [here](https://github.com/rust-lang/miri/pull/2720#issuecomment-1347867264). Instead you should
- support all modes on a Unix host, just forwarding the flag
- throw_unsup on a Windows host for non-default flags

Reply:
My mistake, I confused the host with the target. Hopefully fixed now, and also setting the mode on `options` too